predmap.py

```diff
@@ -274,9 +285,26 @@ def set_rasters_to_column(self):
                 band_np = band_np.astype(float)
                 # mark NaNs
                 band_np[band.GetMaskBand().ReadAsArray() == 0] = np.nan
-
                 feats.append(np.reshape(band_np, (-1, )))
 
+                # Separate rasters with single and multilayer
+                if raster.RasterCount == 1:
+                    path = self.fnames_features[aux]
+                    colname = path.split('\\')[-1].split('.')[0]
```

Comment:
Better to use the `Path` commands. Something like:
````python
Path(path).resolve().stem
````

Also, the  outer loop can be zipped so no `aux` variables are necessary, maybe something like

````python
        for raster, raster_name in zip(self.feature_rasters, self.fnames_features):
             ...
````

Reply:
Added the new outer loop and the suggestion of Path  to get the colnames. The code is more clean now1